distributed/comm/tcp.py

```diff
@@ -436,9 +436,17 @@ async def connect(self, address, deserialize=True, **connection_args):
         kwargs = self._get_connect_args(**connection_args)
 
         try:
-            stream = await self.client.connect(
-                ip, port, max_buffer_size=MAX_BUFFER_SIZE, **kwargs
-            )
+            # server_hostname option (for SNI) only works with tornado.iostream.IOStream
+            if "server_hostname" in kwargs:
+                plain_stream = await self.client.connect(
```

Comment:
I think it's actually better to reassign `stream` here rather than have two variables, this is so the `stream.closed()` logic below can handle both 

Another option would be to chain these as `stream = (await self.client.connect(...)).start_tls(...)`